Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Ride/EndRide/Internal.hs

```diff
@@ -154,8 +154,10 @@ endRideTransaction driverId booking ride mbFareParams mbRiderDetailsId newFarePa
             then do
               logWarning "Unable to calculate dispute chances used"
               return 0
-            else return $ round $ newFareParams.customerCancellationDues / thresholdConfig.cancellationFee
+            else do
+              return $ round $ newFareParams.customerCancellationDues / thresholdConfig.cancellationFee
```

Comment:
According to me, if there is any previous due, we should take the amount and reset dues and `disputeChancesUsed` both to zero.

Reply:
No, we can't do this. See this case
Customer cancelled 3 times so cancellation Dues = 30 
Disputes this amount so dispute chances used = 30
Cancels 1 time so cancellationDue = 10
Now pays it in next ride so dispute chances should be updated to 2 not to 0